data-access/RoiRepository.cs

```diff
@@ -0,0 +1,296 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Dapper;
+using Microsoft.Extensions.Configuration;
+using SQE.SqeHttpApi.DataAccess.Helpers;
+using SQE.SqeHttpApi.DataAccess.Models;
+using SQE.SqeHttpApi.DataAccess.Queries;
+
+namespace SQE.SqeHttpApi.DataAccess
+{
+	public interface IRoiRepository
+	{
+		Task<List<SignInterpretationROI>> CreateRoisAsync(EditionUserInfo editionUser,
+			List<SetSignInterpretationROI> newRois);
+
+		Task<List<UpdatedSignInterpretationROI>> UpdateRoisAsync(EditionUserInfo editionUser,
+			List<SignInterpretationROI> updateRois);
+
+		Task DeletRoisAsync(EditionUserInfo editionUser, List<uint> deleteRoiIds);
+
+		Task<DetailedSignInterpretationROI> GetSignInterpretationRoiByIdAsync(EditionUserInfo editionUser,
+			uint signInterpretationRoiId);
+
+		Task<List<DetailedSignInterpretationROI>> GetSignInterpretationRoisByArtefactIdAsync(
+			EditionUserInfo editionUser,
+			uint artefactId);
+	}
+
+	public class RoiRepository : DbConnectionBase, IRoiRepository
+	{
+		private readonly IDatabaseWriter _databaseWriter;
+
+		public RoiRepository(IConfiguration config, IDatabaseWriter databaseWriter) : base(config)
+		{
+			_databaseWriter = databaseWriter;
+		}
+
+		/// <summary>
+		/// Creates a sign interpretation roi from a list.
+		/// </summary>
+		/// <param name="editionUser">UserInfo object with user details and edition permissions</param>
+		/// <param name="newRois">List of rois to be added to the system.</param>
+		/// <returns></returns>
+		public async Task<List<SignInterpretationROI>> CreateRoisAsync(EditionUserInfo editionUser,
+			List<SetSignInterpretationROI> newRois)
+		{
+			return (await Task.WhenAll(newRois.Select(
+				async (x) =>
+				{
+					var roiShapeId = CreateRoiShapeAsync(x.Shape);
```

Comment:
`roiShapeId` is not really the shape id, it's a task that will eventually resolve the shape id. I think the name should be `roiShapeIdTask`. I was going to call this a bug, but then I saw the `await` a few lines down. So it's not a bug, just a variable with a confusing name.